src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -73,7 +75,16 @@ public function normalize(mixed $object, string $format = null, array $context =
             $object = $object->setTimezone($timezone);
         }
 
-        return $object->format($dateTimeFormat);
+        return match ($context[self::TIMESTAMP_CAST_KEY] ?? $this->defaultContext[self::TIMESTAMP_CAST_KEY]) {
```

Comment:
Instead of nesting matches, I'd do:
```php
if ($context[self::TIMESTAMP_CAST_KEY] ?? $this->defaultContext[self::TIMESTAMP_CAST_KEY]) {
    return match ($dateTimeFormat) {
        'U' => $object->getTimestamp(),
        'U.u' => (float) $object->format('U.u'),
        'Uv' => (int) $object->format('Uv'),
        'Uu' => (int) $object->format('Uu'),
       default => $object->format($dateTimeFormat),
    }
}

return $object->format($dateTimeFormat);
```